src/Service.Tests/SqlTests/GraphQLMutationTests/DwSqlGraphQLMutationTests.cs

```diff
@@ -0,0 +1,604 @@
+
+        #region  Positive Tests
+
+        /// <summary>
+        /// <code>Do: </code> Inserts new book and return its id and title
+        /// <code>Check: </code> If book with the expected values of the new book is present in the database and
+        /// if the mutation query has returned the correct information
+        /// </summary>
+        [TestMethod]
+        public async Task InsertMutation()
+        {
+            string graphQLMutationName = "createbook";
+            string graphQLMutation = @"
+                mutation {
+                    createbook(item: { id: 1, title: ""My New Book"", publisher_id: 1234 }) {
+                        result
+                    }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+            Assert.IsTrue(actual.ToString().Contains("success"), "DwSql mutation insert should be successful");
+        }
+
+        /// <summary>
+        /// <code>Do: </code> Inserts new book using variables to set its title and publisher_id
+        /// <code>Check: </code> If book with the expected values of the new book is present in the database and
+        /// if the mutation query has returned the correct information
+        /// </summary>
+        [TestMethod]
+        public async Task InsertMutationWithVariables()
+        {
+            string graphQLMutationName = "createbook";
+            string graphQLMutation = @"
+                mutation($title: String!, $publisher_id: Int!) {
+                    createbook(item: { id: 2, title: $title, publisher_id: $publisher_id }) {
+                        result
+                    }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true, new() { { "title", "My New Book" }, { "publisher_id", 1234 } });
+            Assert.IsTrue(actual.ToString().Contains("success"), "DwSql mutation insert with variables should be successful");
+        }
+
+        /// <summary>
+        /// <code>Do: </code>Inserts a new book in database and returns the typename
+        /// <code>Check: </code>if the mutation executed successfully and returned the correct typename
+        /// </summary>
+        [TestMethod]
+        public override async Task InsertMutationWithOnlyTypenameInSelectionSet()
+        {
+            string graphQLMutationName = "createbook";
+            string graphQLMutation = @"
+                mutation {
+                    createbook(item: { id: 1, title: ""Awesome Book"", publisher_id: 1234 }) {
+                        __typename
+                    }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+            string expected = @"
+              {
+                ""__typename"": ""DbOperationResult""
```

Comment:
this needs to be documented because the typename of the result does not match the object type that the mutation affects. I think you've mentioned this in design discussion as point of confusion for developers because every mutation would return the same type.

Reply:
Do you mean just on the test or there is some specific place wherin we need to add this documentation?

Reply:
Ok, where do i get access to add that?

Reply:
Will create issue in docs repo.